services/chaospod.go

```diff
@@ -590,41 +622,6 @@ func (m *chaosPodService) removeFinalizerForChaosPod(ctx context.Context, chaosP
 	return nil
 }
 
-func (m *chaosPodService) isFinalizerNotRemovableForChaosPod(chaosPod *corev1.Pod) bool {
-	switch chaosPod.Status.Phase {
-	case corev1.PodSucceeded, corev1.PodPending:
-		// we can remove the pod and the finalizer, so that it'll be garbage collected
-		return false
-	case corev1.PodFailed:
-		// we need to determine if we can remove it safely or if we need to block disruption deletion
-		// check if a container has been created (if not, the disruption was not injected)
-		if len(chaosPod.Status.ContainerStatuses) == 0 {
-			return false
-		}
-
-		// if the pod died only because it exceeded its activeDeadlineSeconds, we can remove the finalizer
-		if chaosPod.Status.Reason == "DeadlineExceeded" {
-			return false
-		}
-
-		// check if the container was able to start or not
-		// if not, we can safely delete the pod since the disruption was not injected
-		for _, cs := range chaosPod.Status.ContainerStatuses {
-			if cs.Name != "injector" {
-				continue
-			}
-
-			if cs.State.Terminated != nil && cs.State.Terminated.Reason == "StartError" {
-				return false
-			}
-
-			break
-		}
-	}
-
-	return true
```

Comment:
This line has not the same behavior as if we use the default of the switch case?

Reply:
No :(. What happens is this returns "true", telling the caller that the finalizer _cannot_ be removed. The caller then assumes that the chaos pod is stuck on removal. We need a way to distinguish the "not yet" removal, from the "not ever" removal, and rather than complicate the return value of this function, I just in-lined it.